### PR TITLE
Use introspection to assign experiments to workers

### PR DESCRIPTION
**Pull request by marcellp**
# Background
#### Link to issue
https://biomage.atlassian.net/browse/BIOMAGE-1248

#### Link to staging deployment URL
https://ui-marcellp-worker165.scp-staging.biomage.net

#### Links to any Pull Requests related to this
N/A

#### Anything else the reviewers should know about the changes here
N/A

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [x] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [x] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [x] Relevant Github READMEs updated
- [x] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: python/src/worker/config/__init__.py [hunk 1]
```diff
@@ -1,24 +1,22 @@
 import os
 import types
+import re
+from logging import info
```
Comment (ivababukova) [inline python/src/worker/config/__init__.py]: you can remove this import, it is not being used

File: r/work.R [hunk 2]
```diff
@@ -1,5 +1,5 @@
-library(Seurat)
-library(dplyr)
+# library(Seurat)
+# library(dplyr)
```
Comment (ivababukova) [inline r/work.R]: this needs to be added back

File: r/work.R [hunk 3]
```diff
@@ -159,8 +158,27 @@ create_app <- function(last_modified, data, fpath) {
...
+        experiment_id <- labels["experimentId", ]
+    }
+
+    if(is.na(experiment_id)) {
+        experiment_id <- Sys.getenv("EXPERIMENT_ID", unset = NA)
+    }
+    
+    if(is.na(experiment_id)) {
+        message("No experiment ID label set yet, waiting...")
+        Sys.sleep(5)
```
Comment (ivababukova) [inline r/work.R:176]: why is this in a separate if statements that are the same, rather than under 1 if statement?

**Comment by ivababukova** [comment]
Another thing is also how is this going to run locally? I was looking today with @ogibson-biomage at some PRs that are trying to autoload experiment id locally and I recommended him to pair-up with you on reviewing these, because they look very relevant to what you are doing

File: r/work.R (hunk 3 above)
Comment (marcellp) [reply r/work.R:176]: Essentially, we read from the file. If the file does not contain experiment_id, we read the env variable. If env variable also does not contain it, we're waiting.

File: r/work.R (hunk 2 above)
Comment (marcellp) [reply r/work.R]: Thanks, fixed.

File: python/src/worker/config/__init__.py (hunk 1 above)
Comment (marcellp) [reply python/src/worker/config/__init__.py]: Fixed.

**Comment by marcellp** [comment]
> Another thing is also how is this going to run locally? I was looking today with @ogibson-biomage at some PRs that are trying to autoload experiment id locally and I recommended him to pair-up with you on reviewing these, because they look very relevant to what you are doing

Using `EXPERIMENT_ID` as an env variable locally will still work, as it is a fallback in case there is nothing in `/etc/podinfo/labels`. It was done like this by design in order to make it work with local development.

File: r/work.R (hunk 3 above)
Comment (ivababukova) [reply r/work.R:176]: indeed, makes sense. Let's keep it this way

**Review by ivababukova — APPROVED** [review]

**Pull request merged**